python-sdk/tests_integration/astro_deploy/astro_sdk_connections.py

```diff
@@ -0,0 +1,87 @@
+import os
+from datetime import datetime
+from textwrap import dedent
+
+import yaml
+from airflow.decorators import dag, task
+from airflow.models import Connection
+from airflow.utils.db import create_default_connections
+from airflow.utils.session import create_session
+
+connections_yaml = f"{os.environ['AIRFLOW_HOME']}/include/connections_template.yaml"
+
+
+@task
+def replace_env_var_in_yaml():
+    with open(connections_yaml) as file:
+        yaml_content = yaml.safe_load(file)
+
+        # Recursively replace environment variables in the YAML content
+    connections_details = recursive_replace_env_vars(yaml_content)
+
+    # Save the modified YAML content back to the file
+    with open(connections_yaml, "w") as file:
+        yaml.dump(connections_details, file, default_flow_style=False)
+
+
+def recursive_replace_env_vars(data):
+    if isinstance(data, dict):
+        for key, value in data.items():
+            if isinstance(value, str):
+                # Check if the string value is an environment variable
+                if value.startswith("$") and value[1:] in os.environ:
+                    data[key] = os.environ[value[1:]]
+            else:
+                # Recursively replace environment variables in nested data structures
+                data[key] = recursive_replace_env_vars(value)
+    elif isinstance(data, list):
+        # Recursively replace environment variables in lists
+        data = [recursive_replace_env_vars(item) for item in data]
+
+    return data
+
+
+@task
+def create_connections():
+    with open(connections_yaml) as fp:
+        yaml_with_env = os.path.expandvars(fp.read())
```

Comment:
In reference to above comment - https://github.com/astronomer/astro-sdk/pull/1937/files#r1198702725
`yaml_with_env = os.path.expandvars(fp.read())` <- this didn't do the trick for you? we may not need `replace_env_var_in_yam`l task as all

Reply:
Removed this change, will be using secrets backend for connection storage